libraries/stdlib/src/kotlin/collections/Grouping.kt

```diff
@@ -0,0 +1,282 @@
+package kotlin.collections
+
+/**
+ * Represents a source of elements with a [keyOf] function, which can be applied to each element to get its key.
+ *
+ * A [Grouping] structure serves as an intermediate step in group-and-fold operations:
+ * they group elements by their keys and then fold each group with some aggregating operation.
+ *
+ * It is created by attaching `keySelector: (T) -> K` function to a source of elements.
+ * To get an instance of [Grouping] use one of `groupingBy` extension functions:
+ * - [Iterable.groupingBy]
+ * - [Sequence.groupingBy]
```

Comment:
`aggregate` and `aggregateTo` are not intended to be used much in user code, they are here more to implement other folding functions when `fold` or `reduce` do not fit for some reason.
Other functions (fold, reduce, eachCount, etc) will get samples eventually.